config/init_server_creds.go

```diff
@@ -574,9 +799,26 @@ func loadIssuerPublicJWKS(existingJWKS string, issuerKeyFile string) (jwk.Set, e
 // Return the private JWK for the server to sign tokens
 func GetIssuerPrivateJWK() (jwk.Key, error) {
 	key := issuerPrivateJWK.Load()
-	if key == nil {
-		issuerKeyFile := param.IssuerKey.GetString()
-		newKey, err := loadIssuerPrivateJWK(issuerKeyFile)
+	issuerKeysDir := param.IssuerKeysDirectory.GetString()
+	currentIssuerKeysDir := getCurrentIssuerKeysDir()
+	// Handles runtime changes to the issuer keys directory (configured via "IssuerKeysDirectory" parameter).
```

Comment:
Why would runtime changes occur?

The parameter is assigned/parsed on startup. If the server settings are changed in the web-ui it forces a restart. Can this actually occur?

Reply:
Currently runtime changes occur happen in two test scenarios:
"invalid-token-sig-key" in `pelican/web_ui/prometheus_test.go`, "diff-secrets-yield-diff-result" in `pelican/config/encrypted_test.go`